src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java

```diff
@@ -0,0 +1,138 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.mercury;
+
+import java.util.List;
+
+import org.openqa.selenium.ElementNotVisibleException;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+/**
+ * @author Tomasz Napieralski
+ */
+public class TableOfContentPageObject extends MercuryBasePageObject {
+
+  @FindBy(css = "nav.table-of-contents")
+  private WebElement tocAll;
+  @FindBy(css = "section.article-body h2")
+  private List<WebElement> allH2;
+  @FindBy(css = "section.article-body > *")
+  private List<WebElement> articleBodyTags;
+  @FindBy(css = "nav.table-of-contents a")
+  private List<WebElement> listOfLinks;
+  @FindBy(css = "nav.table-of-contents ol")
+  private WebElement TOCMenu;
+
+  public TableOfContentPageObject(WebDriver driver) {
+    super(driver);
+  }
+
+  public boolean isTOCDisplayed() {
+    try {
+      if (tocAll.isDisplayed()) {
+        return true;
+      }
+    } catch (NoSuchElementException e) {
+    }
+    return false;
+  }
+
+  public boolean areH2onPage() {
```

Comment:
In BasePageObject is method boolean checkIfElementOnPage(WebElement element) but it checks just one WebElement not the list of elements.
Two solutions:
1) leave public boolean areH2onPage()
2) add to BasePageObject method boolean checkIfElementsOnPage(List<WebElement> elements)